dace/libraries/blas/nodes/axpy.py

```diff
@@ -115,6 +118,51 @@ def expansion(node, parent_state: SDFGState, parent_sdfg: SDFG, **kwargs):
             **kwargs)
 
 
+@dace.library.expansion
+class ExpandAxpyFpgaHbm(ExpandTransformation):
+    """
+    An implementation of Axpy based on ExpandAxpyFpga that uses HBM. Expects attached
+    arrays to be on HBM and 2 dimensional inputs/outputs. 
+    """
+    environments = []
+
+    @staticmethod
+    def expansion(node,
+                  parent_state: SDFGState,
+                  parent_sdfg: SDFG,
+                  param="k",
+                  **kwargs):
+        """
+        :param node: Node to expand.
+        :param parent_state: State that the node is in.
+        :param parent_sdfg: SDFG that the node is in.
+        :param param: The param symbol for the top-level map mapping over HBM-banks
+        """
+        sdfg: SDFG = ExpandAxpyFpga.expansion(node, parent_state, parent_sdfg,
+                                              **kwargs)
+
+        desc_x = parent_sdfg.arrays[parent_state.in_edges(node)[0].data.data]
+        desc_y = parent_sdfg.arrays[parent_state.in_edges(node)[1].data.data]
+        desc_res = parent_sdfg.arrays[parent_state.out_edges(node)[0].data.data]
+        banks_x = fpga.parse_location_bank(desc_x)
+        banks_y = fpga.parse_location_bank(desc_y)
+        banks_res = fpga.parse_location_bank(desc_res)
+        tmp_bank_c = fpga.get_multibank_ranges_from_subset(banks_x[1], sdfg)
+        bank_count = tmp_bank_c[1] - tmp_bank_c[
+            0]  # We know this is equal for all arrays it's checked in validation
+
+        from dace.transformation.dataflow import hbm_transform  # Avoid import loop
+        xform = hbm_transform.HbmTransform(sdfg.sdfg_id, -1, {}, -1)
+        xform.outer_map_range = (param, f"0:{bank_count}")
+        xform.update_array_access = ["_x", "_y", "_res"]
+        xform.update_array_banks = [("_x", banks_x[0], banks_x[1]),
+                                    ("_y", banks_y[0], banks_y[1]),
+                                    ("_res", banks_res[0], banks_res[1])]
+        xform.apply(sdfg)
```

Comment:
I think it's not required here, since apply is directly called, which does not append the transformation. (save is only an option when using apply_to)